File: Php/app/Urls/BaseClasses/CashgameUrlModel.php
namespace app\Urls\BaseClasses{

	use app\UrlFormatter;
	use entities\Cashgame;
	use entities\Homegame;

	class CashgameUrlModel extends UrlModel{

		public function __construct($format, Homegame $homegame, Cashgame $cashgame){
			$url = UrlFormatter::formatCashgame($format, $homegame, $cashgame);
			parent::__construct($url);
		}

	}

}

File: Web/Models/Url/CashgameUrlModel.cs
using Core.Classes;
using Web.Models;
using Web.Models.Url;

namespace app{
    public class CashgameUrlModel : UrlModel{

		public CashgameUrlModel(string format, Homegame homegame, Cashgame cashgame)
        {
            SetUrl(UrlFormatter.FormatCashgame(format, homegame, cashgame));
		}

	}

}